include/wx/event.h

```diff
@@ -3522,7 +3522,7 @@ struct WXDLLIMPEXP_BASE wxEventTableEntry : public wxEventTableEntryBase
     const int& m_eventType;
 
 private:
-    wxDECLARE_NO_ASSIGN_CLASS(wxEventTableEntry);
+    wxDECLARE_NO_ASSIGN_DEF_COPY(wxEventTableEntry);
```

Comment:
This is the only one I have some doubts about, as I'm not sure whether we really should be copy-constructing objects of this type at all. Would you have the error message with the warning for it, to show where exactly is its copy ctor used? I can't find it easily by just looking at the code.

Reply:
The copy ctor is rather hidden.

Here is where it is getting used in the code, in auidemo.cpp:
```
wxBEGIN_EVENT_TABLE(wxSizeReportCtrl, wxControl)
    EVT_PAINT(wxSizeReportCtrl::OnPaint) <----
    EVT_SIZE(wxSizeReportCtrl::OnSize)
    EVT_ERASE_BACKGROUND(wxSizeReportCtrl::OnEraseBackground)
wxEND_EVENT_TABLE()
```

The EVT_PAINT macro entry effectively says "add an wxEventTableEntry to the table".  So that's an implicit copy constructor from the warnings:

```
In file included from /Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/samples/aui/auidemo.cpp:12:
In file included from /Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/wxprec.h:42:
In file included from /Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/wx.h:24:
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/event.h:3525:5: warning: definition of implicit copy constructor for 'wxEventTableEntry' is deprecated because it has a user-declared copy assignment operator [-Wdeprecated-copy]
    wxDECLARE_NO_ASSIGN_CLASS(wxEventTableEntry);
    ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/defs.h:3053:20: note: expanded from macro 'wxDECLARE_NO_ASSIGN_CLASS'
        classname& operator=(const classname&) wxMEMBER_DELETE
                   ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/samples/aui/auidemo.cpp:253:5: note: in implicit copy constructor for 'wxEventTableEntry' first required here
    EVT_PAINT(wxSizeReportCtrl::OnPaint)
    ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/event.h:4507:26: note: expanded from macro 'EVT_PAINT'
#define EVT_PAINT(func)  wx__DECLARE_EVT0(wxEVT_PAINT, wxPaintEventHandler(func))
                         ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/event.h:4475:5: note: expanded from macro 'wx__DECLARE_EVT0'
    wx__DECLARE_EVT1(evt, wxID_ANY, fn)
    ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/event.h:4473:5: note: expanded from macro 'wx__DECLARE_EVT1'
    wx__DECLARE_EVT2(evt, id, wxID_ANY, fn)
    ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/event.h:4471:5: note: expanded from macro 'wx__DECLARE_EVT2'
    wxDECLARE_EVENT_TABLE_ENTRY(evt, id1, id2, fn, nullptr),
    ^
/Users/richardpowell/Development/github.com/rmpowell77/wxWidgets.git/include/wx/event.h:87:5: note: expanded from macro 'wxDECLARE_EVENT_TABLE_ENTRY'
    wxEventTableEntry(type, winid, idLast, wxNewEventTableFunctor(type, fn), obj)
    ^
```

I'm not sure we could easily make this type of warning an error, say by removing the default copy-ctor -- that would mean ever call for wxDECLARE_EVENT_TABLE_ENTRY would be an error.

I guess looking at it, while we shouldn't be generally copy-constructing `wxEventTableEntry`, by the way the macros set up the table we effectively are.

Reply:
Thanks!

So it's initializing `sm_eventTableEntries` which uses the copy ctor. I guess this could be avoided by using braces instead in `wxDECLARE_EVENT_TABLE_ENTRY` definition but I'm not sure if it doesn't have some other undesirable side effects, so let's just merge this PR as is, finally.

Thanks again!